code/hud/hudtarget.cpp

```diff
@@ -6716,11 +6772,11 @@ void HudGaugeWeapons::render(float /*frametime*/, bool config)
 			}
 		} else {
 			// just print the weapon's name since this isn't armed
-			renderString(x + fl2i(Weapon_sname_offset_x * scale), name_y, i ? EG_WEAPON_S1 : EG_WEAPON_S2, weapon_name, scale, config);
+			renderString(x + fl2i(Weapon_sname_offset_x * scale), name_y, i ? EG_WEAPON_S1 : EG_WEAPON_S2, weapon_name.c_str(), scale, config);
 		}
 
-		if (!config && !Weapon_info[sw->secondary_bank_weapons[i]].wi_flags[Weapon::Info_Flags::SecondaryNoAmmo]) {
-			int ammo=sw->secondary_bank_ammo[i];
+		if (config || !Weapon_info[sw->secondary_bank_weapons[i]].wi_flags[Weapon::Info_Flags::SecondaryNoAmmo]) {
+			int ammo = config ? 13 : sw->secondary_bank_ammo[i];
```

Comment:
So the gauge assumes ammo of 13 if no ammo is specified?

Everything else seems to make sense.

Reply:
Yes. Just because this is hud config or in other words, just a representation. I didn't think it was terribly worth it to try to calculate a real value or muddy up the table with more flags just for the value.